chore: move HIPAA checks from org to project level

staaldraad opened the pull request:
HIPAA orgs can mark specific projects as sensitive. Those projects will have HIPAA restrictions applied to them. This opens non-HIPAA projects in the same Organization up to use previously disabled features.

## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Feature update

## What is the current behavior?

HIPAA restrictions apply at the Org level for some features, but at the project level for others.

## What is the new behavior?

HIPAA restrictions are now at the project level

## Additional context

https://supabase.com/docs/guides/platform/hipaa-projects

staaldraad committed:
chore: move HIPAA checks from org to project level

HIPAA orgs can mark specific projects as sensitive. Those projects will
have HIPAA restrictions applied to them. This opens non-HIPAA projects
in the same Organization up to use previously disabled features.

staaldraad commented on apps/studio/components/ui/AIAssistantPanel/AIAssistant.tsx:
```diff
@@ -131,8 +133,6 @@ export const AIAssistant = ({ className }: AIAssistantProps) => {
   const currentSchema = searchParams?.get('schema') ?? 'public'
   const currentChat = snap.activeChat?.name
 
-  const { ref } = useParams()
-  const org = useSelectedOrganization()
```
looked like these were duplicated calls to `useParams()` and `useSelectedOrganization()`

staaldraad committed:
fix: don't call react hooks conditionally

kevcodez commented on apps/studio/components/ui/AIAssistantPanel/AIAssistant.tsx:
```diff
@@ -91,7 +92,7 @@ export const AIAssistant = ({ className }: AIAssistantProps) => {
   const isOptedInToAI = useOrgOptedIntoAi()
   const selectedOrganization = useSelectedOrganization()
-  const { id: entityId } = useParams()
+  const { ref: ref, id: entityId } = useParams()
```
```suggestion
  const { ref, id: entityId } = useParams()
```

staaldraad committed:
chore: disable AI in HIPAA

If in a project and the project is HIPAA enabled, disable AI. Otherwise
rely on the opt-in to AI at the org level
